internal/internal_activity.go

```diff
@@ -394,19 +394,16 @@ func (a *activityEnvironmentInterceptor) RecordHeartbeat(ctx context.Context, de
 	}
 	var data *commonpb.Payloads
 	var err error
-	// We would like to be a able to pass in "nil" as part of details(that is no progress to report to)
+	// We would like to be able to pass in "nil" as part of details(that is no progress to report to)
 	if len(details) > 1 || (len(details) == 1 && details[0] != nil) {
 		data, err = encodeArgs(getDataConverterFromActivityCtx(ctx), details)
 		if err != nil {
 			panic(err)
 		}
 	}
 
-	err = a.env.serviceInvoker.Heartbeat(ctx, data, false)
-	if err != nil {
-		log := GetActivityLogger(ctx)
-		log.Debug("RecordActivityHeartbeat with error", tagError, err)
```

Comment:
IIRC, using `GetActivityLogger` from the context has all sorts of tags about the specific activity. I am concerned the new logging statement is not activity specific. Can you confirm what this log looks like before and after this PR?

Reply:
I can use GetActivityLogger method. I was looking at the activityTaskHandler and it seems that it uses its own logger and add tags.

Reply:
:+1: I haven't walked the code in detail, but whatever gets all those activity-specific tags on the logger I think is most reasonable.